chore: merge develop into 1.9.0 release

NoelStephensUnity opened the pull request:
Merging #2845, #2871, #2872, and #2874 into release/1.9.0


## Changelog
NA

## Testing and Documentation

- No tests have been added.
- No documentation changes or additions were necessary.

HenrikPoulsen committed:
Use slough artifactory in CI Part 2 (#2867)

Co-authored-by: Sourcegraph <batch-changes@sourcegraph.com>

NoelStephensUnity committed:
fix: NetworkTransform synchronize position when half float precision is enabled (#2845)

* fix

Edge case fix for half float precision not getting the proper initial base offset.

* update

adding change log entry

* update

adding PR number

* test

updating some of the tests to reduce the verbosity of the debug output.

NoelStephensUnity committed:
chore: replicate PR-2833 physics2d rigidbody2d fix (#2871)

* Fix physics2d not working without physics

* update

updating changelog

---------

Co-authored-by: Depri0n <151538332+Depri0n@users.noreply.github.com>
Co-authored-by: Deprion <Depri0n@yandex.ru>

NoelStephensUnity committed:
chore: migrate PR-2812 remove animator requirement (#2872)

* Do not require component Animator on NetworkAnimator.

* update

Adding changelog entry.

---------

Co-authored-by: jeffgamedev <jbrooks@gala.games>
Co-authored-by: Jeff Brooks <jeph.brooks@gmail.com>

NoelStephensUnity committed:
Merge branch 'develop' into chore/merge-develop-into-1-9-0-release

unity-cla-assistant commented:
[![CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/pull/badge/not_signed)](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.netcode.gameobjects?pullRequest=2873) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.netcode.gameobjects?pullRequest=2873) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: NoelStephensUnity<br/>:x: HenrikPoulsen<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.netcode.gameobjects?pullRequest=2873) it.</sub>

NoelStephensUnity committed:
chore: migrate 2830 allow null references in NetworkBehaviourReference and NetworkObjectReference (#2874)

* initial commit

* update

adding changelog entry

* update and style

Fixing some style issues along with property type issues.

* test

Adding some generalized testing to validate both NetworkBehaviourReference and NetworkObjectReference can be created using a null and serialized when null.

* update

Updating the changed description.

---------

Co-authored-by: Simone Guggiari <simogecko@gmail.com>

NoelStephensUnity committed:
Merge branch 'develop' into chore/merge-develop-into-1-9-0-release
